File: 0205-isomorphic-strings/0205-isomorphic-strings.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        unordered_map<char, char> m, m2;
        for(int i=0; i<s.size(); i++){
            if(m.find(s[i])!=m.end()){
                if(m[s[i]] != t[i])
                    return false;
            }            
            else
                m[s[i]] = t[i];
        }
        
        for(int i=0; i<t.size(); i++){
            if(m2.find(t[i])!=m2.end()){
                if(m2[t[i]] != s[i])
                    return false;
            }
                
            else
                m2[t[i]] = s[i];
        }
        return true;
    }
};

File: 0205-isomorphic-strings/0205-isomorphic-strings.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        d = {}
        n = len(s)
        flag = False
        for i in range(n):
            if(s[i] not in d and t[i] not in d.values()):
                d[s[i]] = t[i]
            elif(s[i] in d and t[i] != d[s[i]]):
                return False
            elif(s[i] not in d and t[i] in d.values()):
                return False
        return True